### PR TITLE
Fix import cycle error/code duplication

### PR DESCRIPTION
**Pull request by sudswasavi**
This commit addresses the issues with import cycle errors that were
caused due to the queue implementation. As a consequence to that the
queue had to be replicated in each layer. With this present model,
the queue package resides in the utils package and is used across
layers.

This also fixes some code review comments around package naming
conventions.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -83,10 +84,11 @@ func main() {
 	c.Start(stopCh)
 
 	// start the go routines draining the queues in various layers
-	ingestionQueue := k8s.SharedWorkQueue().GetQueueByName(utils.ObjectIngestionLayer)
-
+	ingestionQueue := utils.SharedWorkQueue().GetQueueByName(utils.ObjectIngestionLayer)
+	ingestionQueue.SyncFunc = SyncFromIngestionLayer
```
Comment (monotosh-avi) [inline main.go]: Now syncFunc is assigned outside SharedWorkQueue initialiser, can we put a check in Run(), to make sure syncFunc has been added, before running workers ?

File: main.go [hunk 2]
```diff
@@ -99,3 +101,18 @@ func init() {
...
+	// Let's route the key to the graph layer.
+	// NOTE: There's no error propagation from the graph layer back to the workerqueue. We will evaluate
+	// This condition in the future and visit as needed. But right now, there's no necessity for it.
+	//sharedQueue := SharedWorkQueueWrappers().GetQueueByName(queue.GraphLayer)
+	nodes.DequeueIngestion(key)
+	return nil
+}
+
+func SyncFromNodesLayer(key string) error {
+     //	avirest.DeQueueNodes(key)
```
Comment (monotosh-avi) [inline main.go]: We can remove the unused codes.

File: main.go (hunk 1 above)
Comment (sudswasavi) [reply main.go]: Done.

File: main.go (hunk 2 above)
Comment (sudswasavi) [reply main.go]: Done.

File: pkg/utils/avi_obj_cache.go [hunk 3]
```diff
@@ -16,34 +16,39 @@ package utils
...
 	c.PgCache = NewAviCache()
 	c.PoolCache = NewAviCache()
 	c.SvcToPoolCache = NewAviMultiCache()
 	return &c
 }
 
+var cacheInstance *AviObjCache
+var cacheOnce sync.Once
+
+func SharedCacheFetch() *AviObjCache {
```
Comment (adityats) [inline pkg/utils/avi_obj_cache.go]: let's rename this to SharedAviObjCache(). Prefer to not have verbs in variable names.

Comment (sudswasavi) [reply pkg/utils/avi_obj_cache.go]: Done.

**Pull request merged**